### PR TITLE
[CLEANUP] Switch emailing in the EmailService from oelib mail functions to the Core SwiftMailer

### LINKED ISSUE #613: Switch emailing in the EmailService from oelib mail functions to the Core SwiftMailer

### PR DESCRIPTION
**Pull request by Konafets**
Closes #613

### CODE REVIEW

**Comment by Konafets** [comment]
This PR is ready for review.

File: Tests/LegacyUnit/Service/EmailServiceTest.php [hunk 1]
```diff
@@ -443,13 +467,134 @@ public function sendEmailToAttendeesForOrganizerWithoutFooterNotAppendsFooterSep
...
+        $mockedInstance = $this->createMock(\stdClass::class);
+        $mockedClassName = \get_class($mockedInstance);
+        $this->addMockedInstance($mockedClassName, $mockedInstance);
+
+        $this->purgeMockedInstances();
+
+        self::assertNotSame($mockedInstance, GeneralUtility::makeInstance($mockedClassName));
+    }
+
+    private function createMailMessageMock(): MailMessage
```
Comment (oliverklee) [inline Tests/LegacyUnit/Service/EmailServiceTest.php:591]: Please see my review of #675 about mocking the `send` method.

File: Tests/LegacyUnit/Service/EmailServiceTest.php [hunk 2]
```diff
@@ -443,13 +467,134 @@ public function sendEmailToAttendeesForOrganizerWithoutFooterNotAppendsFooterSep
...
+        // manually purge the Typo3 FIFO here, as purgeMockedInstances() is not tested for that yet
+        GeneralUtility::makeInstance($mockedClassName);
+
+        self::assertEmpty($this->mockedClassNames);
+    }
+
+    /**
+     * @test
+     */
+    public function purgeMockedInstancesRemovesInstanceFromTypo3InstanceBuffer()
```
Comment (oliverklee) [inline Tests/LegacyUnit/Service/EmailServiceTest.php:580]: Please see my review of #675 about moving this to a trait.

File: Tests/LegacyUnit/Service/EmailServiceTest.php [hunk 3]
```diff
@@ -443,13 +467,134 @@ public function sendEmailToAttendeesForOrganizerWithoutFooterNotAppendsFooterSep
...
+    /*
+     * Tests for the utility functions
+     */
+
+    /**
+     * @test
+     */
+    public function mockedInstancesListIsEmptyInitially()
+    {
+        self::assertEmpty($this->mockedClassNames);
```
Comment (oliverklee) [inline Tests/LegacyUnit/Service/EmailServiceTest.php:529]: Please be as specific as possible in the tests: `self::assertSame([], …)`

File: Classes/Service/EmailService.php [hunk 4]
```diff
@@ -55,27 +55,24 @@ public function __construct()
...
-            /** @var Mail $eMail */
-            $eMail = GeneralUtility::makeInstance(Mail::class);
-            $eMail->setSender($event->getEmailSender());
-            $eMail->setReplyTo($event->getFirstOrganizer());
-            $eMail->addRecipient($user);
-            $eMail->setSubject($this->replaceMarkers($subject, $event, $user));
-
-            $eMail->setMessage($this->buildMessageBody($body, $event, $user));
+            $sender = $event->getEmailSender();
+            $firstOrganizer = $event->getFirstOrganizer();
```
Comment (oliverklee) [inline Classes/Service/EmailService.php:66]: Let's rename this so the role is more clear: `$firstOrganizer` -> `$replyTo`

**Review by oliverklee — CHANGES_REQUESTED** [review]

**Comment by oliverklee** [comment]
I have created the two traits for you to use: #683

**Review by oliverklee — CHANGES_REQUESTED** [review]
The same changes as in the other PR apply.

**Review by oliverklee — APPROVED** [review]

**Pull request merged**